File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/domain/model/EventType.java
package at.ac.tuwien.sepm.groupphase.backend.domain.model;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.Arrays;

@Getter
@AllArgsConstructor
public enum EventType {

    CONCERTS("CONCERTS"),
    CULTURE("CULTURE"),
    SPORTS("SPORTS"),
    MUSICAL("MUSICAL"),
    SHOW("SHOW"),
    CABARET("CABARET"),
    COMEDY("COMEDY"),
    OTHER("OTHER");

    private final String code;

    /**
     * converts the String Code into the EventType from the enumeration
     * @param code the string describing the EventType
     * @return the ENUM describing the EventType
     */
    public static EventType fromCode(String code) {
        return Arrays.stream(values())
            .filter(eventType -> eventType.getCode().equals(code))
            .findFirst()
            .orElse(null);
    }


}

File: frontend/src/app/components/event-type-select/event-type.ts
export interface EventType {
    displayName: string;
    value: string;
}

export const EventTypes: EventType[] = [
    {displayName: 'Concerts', value: 'CONCERTS'},
    {displayName: 'Culture', value: 'CULTURE'},
    {displayName: 'Sports', value: 'SPORTS'},
    {displayName: 'Musical', value: 'MUSICAL'},
    {displayName: 'Show', value: 'SHOW'},
    {displayName: 'Cabaret', value: 'CABARET'},
    {displayName: 'Comedy', value: 'COMEDY'},
    {displayName: 'Other', value: 'OTHER'},
];
